addons/field_rations/functions/fnc_update.sqf

```diff
@@ -0,0 +1,114 @@
+/*
+ * Author: mharis001, Glowbal, PabstMirror
+ * Main looping update function for thirst/hunger status. Updates the HUD and applies relevant effects (consequences).
+ *
+ * Arguments:
+ * 0: Next MP sync <NUMBER>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [60] call acex_field_rations_fnc_update
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+// 10 sec (interval) * 100 (percent) / 3600 (sec in hour) = 0.2777778
+#define DESCENT_CONSTANT 0.2777778
+
+params ["_nextMpSync"];
+
+if (isNull ACE_player || {!alive ACE_player}) then {
+    // Hide HUD if player dead or doesn't exist
+    [0] call FUNC(showHud);
+} else {
+    private _thirstStatus = ACE_player getVariable [QGVAR(thirst), 100];
+    private _hungerStatus = ACE_player getVariable [QGVAR(hunger), 100];
+
+    private _workMultiplier = 1;
+    if (vehicle ACE_player == ACE_player && {isTouchingGround ACE_player}) then {
+        private _absSpeed = vectorMagnitude (velocity ACE_player);
+        _workMultiplier = linearConversion [1, 7, _absSpeed, 1, 2, true];
+    };
+
+    private _descentWater = _workMultiplier * DESCENT_CONSTANT / GVAR(timeWithoutWater);
+    private _descentFood = _workMultiplier * DESCENT_CONSTANT / GVAR(timeWithoutFood);
+
+    // Modify water loss based on ace_weather temperature
+    if (["ace_weather"] call ACEFUNC(common,isModLoaded)) then {
+        private _temperature = ((getPosASL ACE_player) select 2) call ACEFUNC(weather,calculateTemperatureAtHeight);
+        private _tempFactor = linearConversion [15, 60, _temperature, 1, 1.5, true];
+        _descentWater = _descentWater * _tempFactor;
+        TRACE_2("Modified water loss based on temperature",_temperature,_tempFactor);
+    };
+
+    // Modify water loss based on ace_medical ivBags
+    if (["ace_medical"] call ACEFUNC(common,isModLoaded)) then {
+        private _ivBags = ACE_player getVariable [QACEGVAR(medical,ivBags), []];
+        if (!(_ivBags isEqualTo []) && {!(ACE_player getVariable [QACEGVAR(medical,isBleeding), false])}) then {
+            // Basic medical is 10x faster in IV change, therefore add extra based on medical level
+            // Unable to get perfect change since update runs every 10s
+            private _changeFactor = [6, 1] select (ACEGVAR(medical,level) >= 2);
```

Comment:
don't need to worry about it now, but in medical rewrite this settings will eventually be swapped for `ivFlowRate`